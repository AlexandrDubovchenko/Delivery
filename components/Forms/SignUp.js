import React from 'react';
import { StyleSheet, View, Text } from 'react-native';
import SubmitButton from './Controlers/SubmitButton'
import { FormInput } from './Controlers/Input';
import { connect } from 'react-redux';
import { signup } from '../../redux/reducers/auth-reducer';
import { reduxForm, Field } from 'redux-form';
import { required} from '../../utilities/validation';

const SignUpForm = (props) => {
    const onSubmit = ({ email, password }) => {
        props.signup(email, password)
    }
    return (
        <View style={styles.form}>
            <Field placeholder='Email/Phone' name="email" component={FormInput} validate={[required]}/>
            <Field placeholder='Password' name="password" password={true} component={FormInput} validate={[required]}/>
            <View style={styles.errorContainer}>
                {props.errorMessage &&
                 <Text style={styles.error}>
                    {props.errorMessage.message}
                </Text>}
            </View>
            <SubmitButton onSubmit={props.handleSubmit(onSubmit)} text={props.text} />
        </View>
    )
}

const SignUpReduxForm = reduxForm({ form: 'signIn' })(SignUpForm);

const styles = StyleSheet.create({
    form: {
        flex: 2,
    },
    error: {
        color: 'red'
    },
    errorContainer: {
        alignSelf: "center"
    }
})

const mapStateToProps = (state) => {

    return {
        errorMessage: state.auth.error
    }
}

export default connect(mapStateToProps, { signup })(SignUpReduxForm)