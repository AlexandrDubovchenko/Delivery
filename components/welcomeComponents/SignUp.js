import React from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';






export const SignUp = () => <TouchableOpacity style = {styles.button}><Text>Sign Up</Text></TouchableOpacity>

const styles = StyleSheet.create({
    button: {
        borderRadius: 50,
        padding: 10,
        backgroundColor: "#FFD700"
      }
    });