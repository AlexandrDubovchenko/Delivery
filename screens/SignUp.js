import React from 'react';
import { StyleSheet, View, Text } from 'react-native';
// eslint-disable-next-line import/no-named-as-default
import SignUpForm from '../components/Forms/SignUp';
import { AuthHeader } from '../components/AuthHeader';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#FFD700',
  },
  text: {
    fontSize: 30,
    alignSelf: 'center',
    color: '#006400',
    flex: 0.5,
  },
});

const SignUpScreen = ({ navigation }) => (
  <View style={styles.container}>
    <AuthHeader navigation={navigation} toggle="SignIn" />
    <Text style={styles.text}>Регистрация</Text>
    <SignUpForm text="Регистрация" />
  </View>
);

export default SignUpScreen;
