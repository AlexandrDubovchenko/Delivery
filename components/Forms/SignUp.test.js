import TestRenderer from 'react-test-renderer';
import  {SignUpForm},  from './SignUp';
import React from 'react';

jest.mock('react-native-gesture-handler', () => {});
jest.mock('@ui-kitten/components', () => {});
jest.mock('react-redux', () => {
  return {
    connect: (mapStateToProps, mapDispatchToProps) => (ReactComponent) => ({
      mapStateToProps,
      mapDispatchToProps,
      ReactComponent
    }),
    Provider: ({ children }) => children
  }
})

test('signUp should render without error', () => {
  const signUpTest = TestRenderer.create(<SignUpForm />);
  expect(signUpTest.toJSON()).toMatchSnapshot();
});