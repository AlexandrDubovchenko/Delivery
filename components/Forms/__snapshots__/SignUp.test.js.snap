// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YourReduxFormComponent renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  >
    <View
      style={
        Object {
          "marginHorizontal": 15,
          "marginVertical": 5,
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#F7F9FC",
                "borderColor": "#E4E9F2",
                "borderRadius": 4,
                "borderWidth": 1,
                "minHeight": 40,
                "paddingHorizontal": 8,
                "paddingVertical": 7,
              },
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "width": "100%",
              },
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            appearance="default"
            editable={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            placeholder="Email/Phone"
            placeholderTextColor="#FFD700"
            rejectResponderTermination={true}
            size="medium"
            status="basic"
            style={
              Array [
                Object {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "normal",
                  "marginHorizontal": 8,
                },
                Object {
                  "flexBasis": "auto",
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
                null,
                Object {
                  "color": "#FFD700",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          style={
            Array [
              Object {
                "marginTop": 4,
              },
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "marginHorizontal": 15,
          "marginVertical": 5,
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#F7F9FC",
                "borderColor": "#E4E9F2",
                "borderRadius": 4,
                "borderWidth": 1,
                "minHeight": 40,
                "paddingHorizontal": 8,
                "paddingVertical": 7,
              },
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "width": "100%",
              },
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            appearance="default"
            editable={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            placeholder="Password"
            placeholderTextColor="#FFD700"
            rejectResponderTermination={true}
            secureTextEntry={true}
            size="medium"
            status="basic"
            style={
              Array [
                Object {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "normal",
                  "marginHorizontal": 8,
                },
                Object {
                  "flexBasis": "auto",
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
                null,
                Object {
                  "color": "#FFD700",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          style={
            Array [
              Object {
                "marginTop": 4,
              },
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
        }
      }
    />
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#006400",
            "borderRadius": 20,
            "fontSize": 10,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 100,
            "opacity": 1,
            "right": 12,
            "width": 140,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFD700",
              "fontSize": 17,
            }
          }
        />
      </View>
    </RNGestureHandlerButton>
  </View>
</View>
`;
